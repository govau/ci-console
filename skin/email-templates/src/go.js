import React from "react";

export const Value = pipeline => `{{${pipeline}}}`;

export const ValueC = ({ pipeline }) => <span>{`{{${pipeline}}}`}</span>;

export const If = ({ pipeline, children }) => (
  <div>
    {`{{if ${pipeline}}}`}
    {children}
    {"{{end}}"}
  </div>
);

export const Range = ({ pipeline, children }) => (
  <div>
    {`{{range ${pipeline}}}`}
    {children}
    {"{{end}}"}
  </div>
);
