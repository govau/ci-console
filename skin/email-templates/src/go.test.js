import React from "react";
import renderer from "react-test-renderer";

import { If, Range, Value, ValueC } from "./go";

test("If › renders without crashing", () => {
  const tree = renderer.create(
    <If pipeline=".field">
      <div>children</div>
    </If>
  );
  expect(tree).toMatchSnapshot();
});

test("Range › renders without crashing", () => {
  const tree = renderer.create(
    <Range pipeline="$i, $w := .widgets">
      <div>children</div>
    </Range>
  );
  expect(tree).toMatchSnapshot();
});

test("Value › renders without crashing", () => {
  expect(Value(".field")).toEqual(`{{.field}}`);
});

test("ValueC › renders without crashing", () => {
  const tree = renderer.create(
    <ValueC pipeline=".field">
      <div>children</div>
    </ValueC>
  );
  expect(tree).toMatchSnapshot();
});
