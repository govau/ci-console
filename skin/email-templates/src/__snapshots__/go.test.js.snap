// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If › renders without crashing 1`] = `
<div>
  {{if .field}}
  <div>
    children
  </div>
  {{end}}
</div>
`;

exports[`Range › renders without crashing 1`] = `
<div>
  {{range $i, $w := .widgets}}
  <div>
    children
  </div>
  {{end}}
</div>
`;

exports[`ValueC › renders without crashing 1`] = `
<span>
  {{.field}}
</span>
`;
